### PR TITLE
Persist selected analysis method in the URL

### PR DESCRIPTION
**Pull request by razvanpapadopol**
Fixes https://github.com/Automattic/abacus/issues/884

This PR adds support for the analysis method to be encoded in the URL state.
* Adds newly selected analysis strategy to URL as a query param 
* Use the `analysis-method` query param to pre-select the analysis method and display a success message as shown below.
* If the value for `analysis-method` query param is not a valid analysis strategy, show an error message as shown below, without crashing the UI.

Screenshot 2023-06-29 at 07 59 31

Screenshot 2023-06-29 at 10 30 18

## How has this been tested?

- Automated tests that cover added/changed functionality
- Manual testing with production data (locally)
- Manual testing with mock data (locally or on Vercel)
- Additional manual testing instructions (please specify):

### CODE REVIEW

File: src/components/explat/experiments/single-view/results/ExperimentResults.tsx [hunk 1]
```diff
@@ -235,9 +237,35 @@ export default function ExperimentResults({
...
+
+  const [strategy, setStrategy] = useState<AnalysisStrategy>(() => {
+    if (initialStrategyFromUrl) {
+      try {
+        const initialStrategy = yup
+          .string()
+          .oneOf(Object.values(AnalysisStrategy))
+          .defined()
+          .validateSync(initialStrategyFromUrl)
+        enqueueSnackbar(`'${Analyses.AnalysisStrategyToHuman[initialStrategy]}' analysis is selected!`, {
```
Comment (jessie-ross) [inline src/components/explat/experiments/single-view/results/ExperimentResults.tsx:255]: I think we can be silent if it works.

File: src/components/explat/experiments/single-view/results/ExperimentResults.tsx [hunk 2]
```diff
@@ -235,9 +237,35 @@ export default function ExperimentResults({
...
+          variant: 'success',
+        })
+        return initialStrategy
+      } catch (e) {
+        enqueueSnackbar(`Selecting '${initialStrategyFromUrl}' analysis failed!`, { variant: 'error' })
```
Comment (jessie-ross) [inline src/components/explat/experiments/single-view/results/ExperimentResults.tsx:260]:
```suggestion
        enqueueSnackbar(`Selecting '${initialStrategyFromUrl}' analysis from query parameter failed!`, { variant: 'error' })
```

**Review by jessie-ross — APPROVED** [review]
Thanks for the quick turnaround on this!

See comments but looks great! 🚢

File: src/components/explat/experiments/single-view/results/ExperimentResults.tsx (hunk 1 above)
Comment (razvanpapadopol) [reply src/components/explat/experiments/single-view/results/ExperimentResults.tsx:255]: I've added this to compensate for a missing visual cue that a method other than the recommended one is selected. People are not usually checking the query params when clicking a link or when sharing a link so any hint in the UI should be useful in this case.

**Pull request merged**